### DISCUSSION TITLE
Add CGNS C++ example to github.com/CGNS

### DISCUSSION THREAD

**Discussion by limska** [CGNS/CGNS#278] [CGNStalk]
Dear cgns users,

We have created an C++ example code to demonstrate using CGNS file format to define User Defined Waves (UDW). The UDW project is a Joint Industry Project (JIP) between Technip USA and Siemens Industry Software. The example has minimal dependencies. 
I have put the code on gitlab.com
[https://gitlab.com/limska/udwexample](https://gitlab.com/limska/udwexample)

Would you be open for us to move the project under github.com/CGNS. I think it is a good starting point for someone wanting to use CGNS with C++. 

Thanks
Sava

**Comment by MicK7** [comment]
I understand you develop a new c++ layer over cgnslib. Is it the case ?
I just want to mentionned the existence of  "cgnscpp" wrapper ( http://cgnspp.sourceforge.net/html/index.html ) that does not seem to be maintained anymore. Is your goal to have a new c++ interface or just to call C function from c++ ?

**Reply by limska** [reply]
This is just an example of using C functions from C++. The wrapper will convert C++ to C. For instance instead of `char *` `std::string` is used. I am not using cgnscpp. The example is a good starting point for using C++ and CGNS. I use [https://conan.io](https://conan.io) to handle package management and CMake as the build system. I have tested the build on both Windows and Linux. It should work on MacOS. In my opinion cgnscpp is an overkill. 

Thanks
Sava

**Comment by brtnfld** [comment]
Looks helpful. I thought it might be a nice addition to the CGNS examples in the CGNS source itself, but it's probably too involved for the typical low-level examples found there. 

I would need to get feedback from the CGNS steering committee before we can make a decision. I'm not sure if it matters that CGNS is zlib licensed and yours is MIT.

I'm assuming that the CI would be changed to use, for example, GitHub actions instead. We could also add a reference to the project if you want to keep it under a separate project. It seems like a useful example to include in our regression tests for CGNS.

**Reply by limska** [reply]
Hi,
Any news from the CGNS steering committee? I have no problem changing the license if needed. The code uses [conan](https://conan.io/) to handle third-party dependencies. 

Regards
Sava Slijepcevic
sava.slijepcevic@siemens.com

**Reply by brtnfld** [reply]
Thanks for the reminder, we will discuss it tomorrow.

**Reply by brtnfld** [reply]
The Committee approved including it as an example. Do you plan on doing a Pull Request?

**Reply by limska** [reply]
Should I add it as a separate repository under the CGNS group or add it under CGNS/CGNS/src/examples?

**Reply by brtnfld** [reply]
If it is was added to examples, then what new external build dependencies would that introduce? It is CMake-only builds, correct?
